### PR TITLE
add constant annotaitons for addon

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: addon/v1alpha1/types_managedclusteraddon.go [hunk 1]
```diff
@@ -245,7 +250,23 @@ const (
...
 
-	// HostedManifestLocationLabelKey is the label key for indicating the cluster that the manifest should be deployed in Hosted
-	// mode.
+	// HostedManifestLocationLabelKey is the label key for indicating the cluster that the manifest should be deployed
+	// in Hosted mode.
+	// Deprecated, will be removed in the future release, please use annotation HostedManifestLocationAnnotationKey from v0.10.0.
 	HostedManifestLocationLabelKey = "addon.open-cluster-management.io/hosted-manifest-location"
+
+	// HostedManifestLocationAnnotationKey is the annotation key for indicating the cluster that the manifest should be
+	// deployed in Hosted mode.
```
Comment (qiujian16) [inline addon/v1alpha1/types_managedclusteraddon.go]: should indicate this annotation is added on which resource?

Comment (zhiweiyin318) [reply addon/v1alpha1/types_managedclusteraddon.go]: added

**Pull request merged**